collated/test/A0141021H.md

```diff
@@ -0,0 +1,537 @@
+        latestSavedWhatNow = new WhatNow(abce.data);
+    }
+
+    @Subscribe
+    private void handleShowHelpRequestEvent(ShowHelpRequestEvent she) {
+        helpShown = true;
+    }
+
+    @Subscribe
+    private void handleJumpToListRequestEvent(JumpToListRequestEvent je) {
+        targetedJumpIndex = je.targetIndex;
+    }
+
+    @Before
+    public void setup() {
+        model = new ModelManager();
+        String tempWhatNowFile = saveFolder.getRoot().getPath() + "TempWhatNow.xml";
+        String tempPreferencesFile = saveFolder.getRoot().getPath() + "TempPreferences.json";
+        logic = new LogicManager(model, new StorageManager(tempWhatNowFile, tempPreferencesFile));
+        EventsCenter.getInstance().registerHandler(this);
+
+        latestSavedWhatNow = new WhatNow(model.getWhatNow()); // last saved assumed to be up to date before.
+        helpShown = false;
+        targetedJumpIndex = -1; // non yet
+    }
+
+    @After
+    public void teardown() {
+        EventsCenter.clearSubscribers();
+    }
+
+    @Test
+    public void execute_invalid() throws Exception {
+        String invalidCommand = "       ";
+        assertCommandBehavior(invalidCommand,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+    }
+
+    /**
+     * Executes the command and confirms that the result message is correct.
+     * Both 'WhatNow' and the 'last shown list' are expected to be empty.
+     * @see #assertCommandBehavior(String, String, ReadOnlyWhatNow, List)
+     */
+    private void assertCommandBehavior(String inputCommand, String expectedMessage) throws Exception {
+        assertCommandBehavior(inputCommand, expectedMessage, new WhatNow(), Collections.emptyList());
+    }
+
+    /**
+     * Executes the command and confirms that the result message is correct and
+     * also confirms that the following three parts of the LogicManager object's state are as expected:<br>
+     *      - the internal WhatNow data are same as those in the {@code expectedWhatNow} <br>
+     *      - the backing list shown by UI matches the {@code shownList} <br>
+     *      - {@code expectedWhatNow} was saved to the storage file. <br>
+     */
+    private void assertCommandBehavior(String inputCommand, String expectedMessage,
+                                       ReadOnlyWhatNow expectedWhatNow,
+                                       List<? extends ReadOnlyTask> expectedShownList) throws Exception {
+        //Execute the command
+        CommandResult result = logic.execute(inputCommand);
+        //Confirm the ui display elements should contain the right data
+        assertEquals(expectedMessage, result.feedbackToUser);
+        if (!inputCommand.contains("find") && !inputCommand.contains("change"))
+            assertEquals(expectedShownList, model.getAllTaskTypeList());
+        
+        //Confirm the state of data (saved and in-memory) is as expected
+        if (!inputCommand.contains("change")) {
```

Comment:
may be split the method to individual methods, each checking a different class of inputs?